src/raster/tests.rs

```diff
@@ -693,3 +693,38 @@ fn test_rasterize() {
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0,]
     );
 }
+
+#[test]
+fn test_color_table() {
+    // Raster containing one band.
+    let dataset = Dataset::open(fixture!("test_color_table.tif")).expect("open failure");
+    assert_eq!(dataset.raster_count(), 1);
+
+    // Band is PaletteIndex.
+    let band = dataset.rasterband(1).expect("rasterband failure");
+    assert_eq!(
+        band.color_interpretation(),
+        ColorInterpretation::PaletteIndex
+    );
+
+    // Color table is RGB.
+    let color_table = band.color_table().unwrap();
+    assert_eq!(
+        color_table.palette_interpretation(),
+        gdal_sys::GDALPaletteInterp::GPI_RGB
+    );
```

Comment:
I'm not sure about outputting a `gdal_sys` type here. That would mean you have to depend on the sys crate to use this API. Maybe we should wrap the `gdal_sys::GDALPaletteInterp::GPI_RGB` into a custom enum like `ColorInterpretation` does.

Reply:
I'll add an enum.

Reply:
:+1: